azurerm/resource_arm_public_ip.go

```diff
@@ -48,22 +49,27 @@ func resourceArmPublicIp() *schema.Resource {
 			"public_ip_address_allocation": {
 				Type:             schema.TypeString,
 				Required:         true,
-				ValidateFunc:     validatePublicIpAllocation,
+				ValidateFunc:     validation.StringInSlice([]string{"static", "dynamic"}, true),
```

Comment:
_minor_ I believe there's an Enum we can use for these values

Reply:
There is! thanks updated